techmatters/hrm
# chore: publish pending index-search jobs on write/update ops for cases [CHI-2725]

## Description
This PR makes usage of the SQS client introduced in https://github.com/techmatters/hrm/pull/644 to publish contacts after creation/updates operations performed.

Same generals considerations as https://github.com/techmatters/hrm/pull/645 apply here.
Another note is that, in this PR, we are introducing the notion of `remove` operation for cases in the index. This is because we have a "cancel" (delete) case operation in HRM, which will in turn trigger a delete-from-index operation on ES.

### Checklist
- [x] [Corresponding issue has been opened](https://tech-matters.atlassian.net/browse/CHI-2702)

### Verification steps
- Make sure that the latest version of `search-index-consumer` lambda is deployed to development (this PR).
- Deploy this code to HRM development.
- Create and/or modify a few cases, add contacts to them, initialize cases and "cancel" (delete) them.
- Confirm by looking at `/aws/lambda/development-hrm-search-index-consumer` cloudwatch logs that the messages are being processed by the lambda consumer.
- Go to ElasticSearch console, and confirm by querying `development-us-east-1-hrm-shared` node that the cases are properly indexed.
  - `GET /acd8a2e89748318adf6ddff7df6948deaf-hrm-cases/_search`


### AFTER YOU MERGE

1. Cut a release tag using the GitHub workflow. Wait for it to complete and notify in the #aselo-deploys Slack channel.
2. Comment on the ticket with the release tag version AND any additional instructions required to configure an environment to test the changes.
3. Only then move the ticket into the QA column in JIRA

You are responsible for ensuring the above steps are completed. If you move a ticket into QA without advising what version to test, the QA team will assume the latest tag has the changes. If it does not, the following confusion is on you! :-P
